model/rulefmt/rulefmt.go

```diff
@@ -136,10 +136,11 @@ func (g *RuleGroups) Validate(node ruleGroups) (errs []error) {
 
 // RuleGroup is a list of sequentially evaluated recording and alerting rules.
 type RuleGroup struct {
-	Name     string         `yaml:"name"`
-	Interval model.Duration `yaml:"interval,omitempty"`
-	Limit    int            `yaml:"limit,omitempty"`
-	Rules    []RuleNode     `yaml:"rules"`
+	Name        string          `yaml:"name"`
+	Interval    model.Duration  `yaml:"interval,omitempty"`
+	QueryOffset *model.Duration `yaml:"query_offset,omitempty"`
```

Comment:
Does this need to be a pointer? For the `Interval` above, we just use the zero value to indicate that it's unset (and the `omitempty` ensures that it's not serialized then).

Reply:
Because this can potentially override the configuration value, we need to differentiate between a zero value (which means override whatever yo have in the config) and no value (which means use the global configuration option). What do you think?

Reply:
I see, right. So for example if you have a 5m offset in the global config, but you want a 0m offset in a specific rule group. Yeah, that makes sense, thanks for the explanation :) :+1: